app/models/risk_condition.rb

```diff
@@ -25,7 +25,8 @@ def self.create_condition_for_course_with_type(course_id, type, params, version)
             "Make sure your request body fits the criteria!"
     end
 
-    options = { course_id: course_id, condition_type: type, parameters: params, version: version }
+    options = { course_id: course_id, condition_type: type, parameters: params.to_hash,
```

Comment:
See PR description for more details. Earlier on, `.to_h` was called on `new_metrics_params` (of class `ActionController::Parameters`) which yielded a `ActiveSupport::HashWithIndifferentAccess`. The params here is also a `ActiveSupport::HashWithIndifferentAccess`, and so we use `.to_hash` to convert it to a normal `Hash` before attempting to save. This is due to a breaking change in `Psych` (used somewhere down the stack), where YAML `safe_load` is the new default, which does not support most classes.